### PR TITLE
protocol/bc: move protocol/tx code here

### PR DESCRIPTION
**Pull request by bobg**
...In anticipation of rewriting validation in terms of txentries objects.

Among other benefits, this allows us to get rid of a lot of `init_test.go` files, yay!

(This is a repeat of https://github.com/chain/chain/pull/705.)

### CODE REVIEW

**Comment by bobg** [comment]
PTAL

**Comment by bobg** [comment]
rebased

**Comment by bobg** [comment]
PTAL

**Comment by bobg** [comment]
PTAL

**Comment by kr** [comment]
EYES

File: protocol/bc/map.go [hunk 1]
```diff
@@ -76,10 +74,36 @@ func mapTx(tx *bc.TxData) (headerID bc.Hash, hdr *header, entryMap map[bc.Hash]e
 				}
 
 				assetID := oldIss.AssetID()
-				b := vmutil.NewBuilder()
-				b = b.AddData(oldIss.Nonce).AddOp(vm.OP_DROP).AddOp(vm.OP_ASSET).AddData(assetID[:]).AddOp(vm.OP_EQUAL)
 
-				nonce = newNonce(program{1, b.Program}, tr)
+				// This is the program
+				//   [PUSHDATA(oldIss.Nonce) DROP ASSET PUSHDATA(assetID) EQUAL]
+				// minus a circular dependency on protocol/vm.
```
Comment (kr) [inline protocol/bc/map.go]: This reminds me I would like to get rid of the vm -> bc dependency. Package vm should instead define one or two small interfaces that bc satisfies. But this seems fine to me for now.

File: protocol/bc/txtransaction.go [hunk 2]
```diff
@@ -1,25 +1,15 @@
...
 
 import (
 	"fmt"
 
 	"chain/crypto/sha3pool"
 	"chain/errors"
-	"chain/protocol/bc"
 )
 
-func init() {
```
Comment (kr) [inline protocol/bc/txtransaction.go]: \o/

**Comment by bobg** [comment]
rebased

**Pull request merged**